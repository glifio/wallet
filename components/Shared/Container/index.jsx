import React, { forwardRef } from 'react'
import Box from '../Box'

export default forwardRef(({ ...props }, ref) => (
  <Box ref={ref} {...props} p={[1, 3, 4]} />
))
