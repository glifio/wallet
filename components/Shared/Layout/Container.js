import styled from 'styled-components'
import { layout, margin } from 'styled-system'

export default styled.div.attrs(props => ({
  maxWidth: props.maxWidth || 13,
  minWidth: props.minWidth || 11,
  ...props
}))`
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex-grow: 2;
  ${layout}
  ${margin}
`
