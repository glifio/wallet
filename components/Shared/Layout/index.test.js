import { render } from '@testing-library/react'
import Container from './Container'

describe('Container', () => {
  test('renders the Container', () => {
    const { container } = render(<Container />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders the Container with the right attributes', () => {
    const { container } = render(<Container border='1' width='auto' mt={4} />)

    const containerEl = container.querySelector('div')
    expect(containerEl).toHaveStyle('border: 1;')
    expect(containerEl).toHaveStyle('width: auto;')
    expect(containerEl).toHaveStyle('margin-top: 32px;')
  })
})
