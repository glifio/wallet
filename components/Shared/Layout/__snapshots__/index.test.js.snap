// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders the Container 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  max-width: 13px;
  min-width: 11px;
}

<div
  class="c0"
/>
`;
